lib/ssh_scan/database/mongo_db.rb

```diff
@@ -0,0 +1,40 @@
+require 'mongo'
+
+Mongo::Logger.logger.level = ::Logger::FATAL
+
+module   SSHScan
+  module Database
+    class MongoDb
+      def initialize(opts = {})
+        name = opts["database"]["name"]
+        socket = "#{opts["database"]["server"]}:#{opts["database"]["port"]}"
+        @db = Mongo::Client.new([socket], :database => name)
+        @scans = @db[:scans]
+      end
+
+      def self.from_config_file(file_string)
+        opts = YAML.load_file(file_string)
+        SSHScan::Database::MongoDb.new(opts)
+      end
+
+      def add_scan(worker_id, uuid, result)
+        @scans.insert_one("uuid" => uuid, "scan" => result,
+                          "worker_id" => worker_id, "scanned_on" => Time.now)
+      end
+
+      def delete_scan(uuid)
+        @scans.delete_one(:uuid => uuid)
+      end
```

Comment:
Just another comment here that in Ruby, this is what we're striving for, an Object class that has various interfaces, but the methods are really short, each to understand and document.  This makes for really maintainable less buggy code :)